kobotoolbox/kpi
# Pin pyxform version with support for `background-geopoint` questions

## Checklist

1. [ ] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [X] Ensure the tests pass
4. [X] Run `./python-format.sh` to make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/main/CONTRIBUTING.md)
5. [X] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [ ] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes
8. [X] Create a testing plan for the reviewer and add it to the Testing section

## Description

Support forms with`background-geopoint`questions which save background GPS location in surveys. 

## Notes

⚠️  This should NOT be reviewed until #5126 is merged. 

Kpi is updated to [support version 2.1.1](https://github.com/kobotoolbox/kpi/pull/5126) of pyxform. This version does not include [recent support for `background-geopoint` questions](https://github.com/XLSForm/pyxform/pull/726) so we are backporting this work into 2.1.1. 

Also note that this pr has a small fix for a missing `notify` import in the formbuilder. 

## Testing

1. Reinstall `dev_requirements.txt` in your shell. 
2. Upload an XLS form with a `background-geopoint` question. ([Simple_background_geo.xlsx](https://github.com/user-attachments/files/17510897/Simple_background_geo.xlsx))
3. Create and deploy the project. 
4. Download the XML and verify it has translated the form correctly. See example [here](https://github.com/XLSForm/pyxform/issues/716#issuecomment-2344817692).
